### PR TITLE
feat: introduce logging and errors packages

### LINKED ISSUE #22: How should we handle logging and errors?
### What problem are you facing?

https://pkg.go.dev/github.com/crossplane/crossplane-runtime@v1.13.0/pkg/errors
https://pkg.go.dev/github.com/crossplane/crossplane-runtime@v1.13.0/pkg/logging

You can think of crossplane-runtime as a "Go SDK" of sorts for parts of Crossplane - i.e. core and providers. We have our own error and logging libraries there. Should Functions use those libraries? Use copies/aliases from this SDK? Use the stdlib or something else?

### How could Crossplane help solve your problem?

We should figure out what we want to do here. I can see a case for sticking closer to the stdlib here given that now Go can do the core things our libraries were originally created to do for us (i.e. wrap errors, and emit structured logs).

Some thoughts:
* Admittedly I personally prefer the "UX" for the crossplane-runtime libraries to the stdlib alternatives.
* If stdlib is good enough for Functions, is it good enough for Crossplane now too?

### PR DESCRIPTION
**Pull request by phisco**
### Description of your changes

Fixes #22.

I proxied the logging package to it's `crossplane-runtime` counterpart, I felt it was ok to do so because of the small interface the original package has, while I directly vendored `errors`, as it felt already mainly a proxy to the stdlib `errors` package and didn't want to add one more indirection layer.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: errors/errors.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package errors is a github.com/pkg/errors compatible API for native errors.
+//
+// Copied over from github.com/crossplane/function-sdk-go/errors for
```
Comment (negz) [inline errors/errors.go:19]: Looks like the wrong URL maybe? I'm guessing this should by c/crossplane-runtime.

File: logging/logging.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+// NewNopLogger returns a Logger that does nothing.
+func NewNopLogger() Logger { return logging.NewNopLogger() }
+
+// NewLogrLogger returns a Logger that is satisfied by the supplied logr.Logger,
+// which may be satisfied in turn by various logging implementations (Zap, klog,
+// etc). Debug messages are logged at V(1).
+func NewLogrLogger(l logr.Logger) Logger {
+	return logging.NewLogrLogger(l)
+}
```
Comment (negz) [inline logging/logging.go:40]: There's a `NewLogger` function in `sdk.go`. What do you think about moving the implementation here, and leaving a convenience alias in `sdk.go`? I think this is similar to what controller-runtime does - they have everything in "specific" packages, but really commonly used things get aliased in their root package.

**Review by negz — APPROVED** [review]
Awesome, thank you!

File: logging/logging.go (hunk 2 above)
Comment (phisco) [reply logging/logging.go:44]: sounds good, done!

**Pull request merged**